src/algebra/order/monoid_lemmas_zero_lt.lean

```diff
@@ -419,6 +421,70 @@ lemma pos_iff_pos_of_mul_pos [pos_mul_reflect_lt α] [mul_pos_reflect_lt α] (ha
   0 < a ↔ 0 < b :=
 ⟨pos_of_mul_pos_left hab ∘ le_of_lt, pos_of_mul_pos_right hab ∘ le_of_lt⟩
 
+lemma mul_le_mul_of_le_of_le'' [pos_mul_mono α] [mul_pos_mono α]
+  (h₁ : a ≤ b) (h₂ : c ≤ d) (a0 : 0 ≤ a) (d0 : 0 ≤ d) : a * c ≤ b * d :=
+(mul_le_mul_left'' h₂ a0).trans $ mul_le_mul_right'' h₁ d0
+
+lemma mul_le_mul_of_le_of_le''' [pos_mul_mono α] [mul_pos_mono α]
+  (h₁ : a ≤ b) (h₂ : c ≤ d) (b0 : 0 ≤ b) (c0 : 0 ≤ c) : a * c ≤ b * d :=
+(mul_le_mul_right'' h₁ c0).trans $ mul_le_mul_left'' h₂ b0
+
+lemma mul_le_of_mul_le_left' [pos_mul_mono α]
+  (h : a * b ≤ c) (hle : d ≤ b) (a0 : 0 ≤ a) :
+  a * d ≤ c :=
+(mul_le_mul_left'' hle a0).trans h
+
+lemma le_mul_of_le_mul_left' [pos_mul_mono α]
+  (h : a ≤ b * c) (hle : c ≤ d) (b0 : 0 ≤ b) :
+  a ≤ b * d :=
+h.trans (mul_le_mul_left'' hle b0)
+
+lemma mul_le_of_mul_le_right' [mul_pos_mono α]
+  (h : a * b ≤ c) (hle : d ≤ a) (b0 : 0 ≤ b) :
+  d * b ≤ c :=
+(mul_le_mul_right'' hle b0).trans h
+
+lemma le_mul_of_le_mul_right' [mul_pos_mono α]
+  (h : a ≤ b * c) (hle : b ≤ d) (c0 : 0 ≤ c) :
+  a ≤ d * c :=
+h.trans (mul_le_mul_right'' hle c0)
+
+lemma mul_left_cancel [pos_mul_mono_rev α]
+  (h : a * b = a * c) (a0 : 0 < a) :
+  b = c :=
```

Comment:
make into an `iff`, like [`add_left_cancel_iff`](https://leanprover-community.github.io/mathlib_docs/algebra/group/defs.html#add_left_cancel_iff). If you want you can also keep the implication, but it might be better to only have the `iff`.